### DISCUSSION TITLE
Continue Keyword Combined With Selection Statement

### DISCUSSION THREAD

**Discussion by CheetahChrome** [dotnet/csharplang#3689] [Language Ideas]
Have the `continue` keyword/operation have the optional processing of a boolean statement to determine action. This would allow this syntax:

```
continue ( A == B );

```
which would replace this construct currently

```
if ( A == B )
   continue;
```

**Comment by HaloFour** [comment]
Doesn't seem like much of a savings.

```cs
continue (A == B);
if (A == B) continue;
```

Why should `continue` be special in this regard?

**Comment by CheetahChrome** [comment]
I feel adding the `continue` with/as a conditional expression provides similar flexibility as the `?:` construct. It provides a localized *readability* of the *what* the continue *is to be used on* in a quicker fashion w/o the 'if' framework.

Unclear to me what you feel a "savings" truly encompass or what complexity one would seek before adding language enhancements. 

---

I prefer the symmetry of conditional expressions over the standard conditionals and adding this, at least to me, seems a valid alternative to specify what the code's `continue`  really jumps on; meaning the definition is not clouded with the `if` construct and clearly visible to the reader of the code.

**Comment by svick** [comment]
@CheetahChrome `if` is not some heavy-weight "framework", it's two characters that make it very clear what the code does (especially if you've used almost any common programming language before). But to me, your proposed syntax does not indicate in any way that the expression after `continue` is a condition, so I think it makes the code much less readable.

The conditional expression (`?:`) is useful because it is an *expression*, which means it gives you flexibility in how you structure your code. But you're proposing to replace one statement with another statement, with no real change in structure, so I don't see how it increases flexibility.

**Comment by theunrepentantgeek** [comment]
See also #724 (Conditional control statements) which suggest a similar syntax but for a wider range of operators - and which is closed.
